app/src/main/java/org/mozilla/fenix/home/recentvisits/view/RecentlyVisited.kt

```diff
@@ -191,6 +199,63 @@ private fun RecentVisitItem(
     }
 }
 
+/**
+ * A recently visited history item.
+ *
+ * @param recentVisit The [RecentHistoryHighlight] to display.
+ * @param showDividerLine Whether to show a divider line at the bottom.
+ * @param onRecentVisitClick Invoked when the user clicks on a recent visit.
+ */
+@Composable
+private fun RecentlyVisitedHighlight(
+    recentVisit: RecentHistoryHighlight,
+    showDividerLine: Boolean,
+    onRecentVisitClick: (RecentHistoryHighlight) -> Unit = { _ -> },
+) {
+    Row(
+        modifier = Modifier
+            .clickable { onRecentVisitClick(recentVisit) }
+            .size(268.dp, 56.dp),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Favicon(url = recentVisit.url, size = 24.dp)
+
+        Spacer(modifier = Modifier.width(16.dp))
+
+        Box(modifier = Modifier.fillMaxSize()) {
+            RecentlyVisitedTitle(
+                text = recentVisit.title,
+                modifier = Modifier.align(Alignment.CenterStart)
+            )
+
+            if (showDividerLine) {
+                RecentlyVisitedDivider(modifier = Modifier.align(Alignment.BottomCenter))
+            }
+        }
+    }
+}
+
+/**
+ * The title of a recent visit.
+ *
+ * @param text [String] that will be display. Will be ellipsized if cannot fit on one line.
+ * @param modifier [Modifier] allowing to perfectly place this.
```

Comment:
```suggestion
 * @param modifier Optional [Modifier] to be applied to the layout.
```
Using one of your previous kDoc description so we hopefully don't keep rewording this differently everywhere.